### ISSUE TITLE
[ENTRY] Neural Networks for QAOA Optimization

### ISSUE THREAD

**Issue by lockwo** [XanaduAI/QHack2021#60]
### Team Name: 

hauntological haberdashery

### Project Description: 

In this project, we evaluate the potential of using classical machine learning to help expedite quantum machine learning. Specifically, we seek to address the challenges of optimizing QAOA. With larger depth (p) and more qubits, the gradients of QAOA become more difficult and costly to evaluate. Our solution is to use a classical neural network to approximate the gradients of the quantum system.

We create a neural network that takes the previous 4 weights and losses as input and outputs the predicted next weights. We generate the training data via quantum gradient descent. This is trained on 4-10 qubits systems with a depth of 10. We then evaluate this as an optimization method by comparing the results of the neural network with quantum gradients on unseen problems inside and outside the training distribution.

### Presentation: 

https://youtu.be/2PtGPYSePd4

### Source code: 

https://github.com/lockwo/QHack-Open

**Comment by co9olguy** [comment]
Thanks for the submission! We hope you have enjoyed participating in QHack :smiley: 

We will be assessing the entries and contacting the winners separately. Winners will be publicly announced sometime in the next month.

We will also be freezing the GitHub repo as we sort through the submitted projects, so you will not be able to update this submission.
